kernel/src/mem.rs

```diff
@@ -18,15 +18,15 @@ pub struct Shared;
 /// Base type for an AppSlice that holds the raw pointer to the memory region
 /// the app shared with the kernel.
 pub struct AppPtr<L, T> {
-    ptr: Unique<T>,
+    ptr: NonNull<T>,
     process: AppId,
     _phantom: PhantomData<L>,
 }
 
 impl<L, T> AppPtr<L, T> {
-    unsafe fn new(ptr: *mut T, appid: AppId) -> AppPtr<L, T> {
+    fn new(ptr: NonNull<T>, appid: AppId) -> AppPtr<L, T> {
```

Comment:
This is internal, so I could go either way on marking it as unsafe. But again it cannot verify the pointer it receives.

Reply:
I think you're right -- I updated this to be unsafe and to document its assumptions.